cpp/csp/engine/Struct.h

```diff
@@ -756,11 +756,15 @@ class Struct
     void decref()
     {
 //Work around GCC12 bug mis-identifying this code as use-after-free
-//#pragma GCC diagnostic push
-//#pragma GCC diagnostic ignored "-Wuse-after-free"
+#ifdef __linux__
```

Comment:
should probably be ifdef __GNUC__ since this is a gcc issue (and also the pragma is for GCC)

Reply:
clang also defines __GNUC__ 
https://stackoverflow.com/questions/38499462/how-to-tell-clang-to-stop-pretending-to-be-other-compilers